### DISCUSSION TITLE
circular buffer assertion failure

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3641] [Q&A] [answered]
In some cases, I am getting a failure of an assertion `used <= mFreeSpace` in `backend::CommandBufferQueue::flush`, and the comment above that line says "circular buffer is too small, we corrupted the stream".  Does this indicate a bug, or am I just asking Filament to render too many distinct renderables?

**Comment by romainguy** [comment, marked as answer]
It's unfortunately the latter, see issue #2507. We do provide a compile-time way to increase the size of the commande buffer queue. You can do this by passing define flags to CMake: `FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB`, `FILAMENT_PER_FRAME_COMMANDS_SIZE_IN_MB` and `FILAMENT_MIN_COMMAND_BUFFERS_SIZE_IN_MB` (which default respectively to 2, 2, and 1 MiB).

**Reply by jwwalker** [reply]
I'm afraid I don't see the relevance of "Clarify Windows build instructions".

It would be nice if Filament could fail more gracefully, or at least allow setting these limits as a run-time initialization.

**Reply by romainguy** [reply]
Sorry it's supposed to be issue #2057.
